universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java

```diff
@@ -59,12 +61,12 @@ public QuestionController(
               String exception = "";
               try {
                 QuestionDefinition definition = questionForm.getBuilder().setVersion(1L).build();
-                boolean success = service.create(definition).isPresent();
-                if (!success) {
-                  exception =
-                      String.format(
-                          "create failed: this is most likely you specify an invalid path %s",
-                          definition.getPath());
+                ErrorAnd<QuestionDefinition, QuestionServiceError> result =
+                    service.create(definition);
+                if (result.isError()) {
+                  for (QuestionServiceError e : result.getErrors()) {
+                    exception += e.message();
```

Comment:
Question (blocking): how do multiple validation messages appear to the user? Are they concatenated into a run-on string? Please share a screenshot.

Reply:
![Screen Shot 2021-03-04 at 11 08 46 AM](https://user-images.githubusercontent.com/2597580/110016812-55417500-7cda-11eb-8b05-c60a98d1fe77.png)
Yeah, it is all one string now.

Reply:
Please add a period and space between the messages e.g.
```
errorMessageJoiner = new StringJoiner(". ", "", ".");

for (errorMessage : messages) {
  errorMessageJoiner.add(errorMessage);
}
errorMessageJoiner.toString();
```

Reply:
Done

It looks like this now.
![Screen Shot 2021-03-04 at 4 03 30 PM](https://user-images.githubusercontent.com/2597580/110047754-4de29180-7d03-11eb-8d52-608256854442.png)